client/client.go

```diff
@@ -32,6 +32,10 @@ const (
 var E2eeSerializers = map[string]serialize.Serialization{"cbor": CBOR}
 var PPTSerializers = map[string]serialize.Serialization{"json": JSON, "msgpack": MSGPACK, "cbor": CBOR}
 
+type invocationCtxKey string
+
+const invocationIDCtxKey invocationCtxKey = "invocationID"
```

Comment:
I think this can be a private type of 0 size to also be size efficient, such as:
```
type invocationCtxKey struct{}
```
There's some mention of it in https://pkg.go.dev/context?utm_source=godoc#WithValue and the relevant discussion in https://github.com/golang/go/issues/17826